### PR TITLE
Doc: Update release notes for master through to 2019-01-01

### PR DESCRIPTION
**Pull request by harding**
This adds release notes for the changes listed by `git log 11e1ac3...fb52d0684 --merges`, picking up where #14688 left off in my attempt to update the release notes every month or so (reducing the amount of work that needs to be done near release time).  I've excluded changes backported to the 0.17 branch (whether they've been released yet or not).

### CODE REVIEW

**Review by fanquake — CHANGES_REQUESTED** [review]

File: doc/release-notes.md [hunk 1]
```diff
@@ -216,6 +285,16 @@ Configuration
   deterministic wallets. This release makes specifying `-usehd` an
   invalid configuration option.
 
+Changes for particular platforms
+--------------------------------
+
+- On MacOS, Bitcoin Core now opts out of application CPU throttling
+  ("app nap") during initial block chain download, when catching up from
```
[the remark this answers is not in the document]
Comment (harding) [reply doc/release-notes.md]: I'm happy to change this, but is this the repository's policy or just your preference?  In previous (but admittedly dated) discussions I recall about this topic, the use of "block chain" (two words) was preferred.  E.g., #1910, #5242, and #6666

[the remark this answers is not in the document]
Comment (fanquake) [reply doc/release-notes.md]: @harding I'm not really fussed either way, was more for consistency with the rest of file. All other uses in here are `blockchain`.

File: doc/release-notes.md [hunk 2]
```diff
@@ -190,8 +218,49 @@ in the Low-level Changes section below.
...
+  previous label of "cold wallet" will be retained.  If you optionally
+  specify any label besides the default when calling `importprivkey`,
+  the new label will be applied to the address.
+
 - See the [Mining](#mining) section for changes to `getblocktemplate`.
 
+Graphical User Interface (GUI)
+------------------------------
+
+- A new Window menu is added alongside the existing File, Settings, and
```
[the remark this answers is not in the document]
Comment (harding) [reply doc/release-notes.md]: "is added" appears to be [grammatically correct](https://ell.stackexchange.com/questions/17132/difference-between-is-added-and-was-added) and is used 28 times in the current source code per `git grep`.  But, if you insist, I'm happy to change it (I prefer it over alternatives for its brevity).

File: doc/release-notes.md [hunk 3]
```diff
@@ -72,16 +72,44 @@ Mining
...
+  nodes. The option will now by default be off for improved privacy and security
+  as well as reduced upload usage. The option can explicitly be turned on for
+  local-network debugging purposes.
+
+- The `rpcallowip` option can no longer be used to automatically listen
+  on all network interfaces.  Instead, the `rpcbind` parameter must also
+  be used to specify the IP addresses to listen on.  Listening for RPC
+  commands over a public network connection is insecure and should be
+  disabled, so a warning is now printed if a user selects such a
+  configuration.  If you need to expose RPC in order to use a tool
```
[the remark this answers is not in the document]
Comment (harding) [reply doc/release-notes.md]: Why?  We don't typically require the use of an article before the name of a protocol, e.g. "If you need to expose {HTTP, SMTP, IMAP, SSL, bittorrent, ...} in order to use..."

**Comment by fanquake** [comment]
Thanks, utACK f3d7d75. Not fussed about the other changes.

**Review by fanquake — APPROVED** [review]
Thanks, utACK f3d7d75. Not fussed about the other changes.

**Comment by laanwj** [comment]
Thanks !!!

ACK f3d7d75e4e80bcd2c6058babb732c9c6cc7522c6

**Comment by MarcoFalke** [comment]
> I've excluded changes backported to the 0.17 branch (whether they've been released yet or not).

Are they mentioned in the release notes on that branch?

**Comment by harding** [comment]
@MarcoFalke I didn't check.  However, I don't believe any of the detached release notes I integrated applied to anything that had been backported to 0.17 so far, so no information should've been lost.  (I did not check pre-0.17 branches.)

The list of noteworthy changes for a typical maintenance release are much smaller than those for a major release, so I don't think the non-master branches would benefit much from periodic release notes updates (but, if you disagree, I'm happy to start opening PRs for that).

If there's anything I can do better, please let me know (or let me know if these updates are annoying and I'll stop).

**Comment by MarcoFalke** [comment]
Oh no, they are very helpful. Usually we had no release notes at all for the whole rc cycle.

What I meant to say is that sometimes release notes get added to the master branch, but not the backport.

E.g. https://github.com/bitcoin/bitcoin/pull/14941/files#diff-c3924c30088c7adb0477128540f35c11R1
and https://github.com/bitcoin/bitcoin/pull/15002/files

So, it would be a pity if the notes get lost due to an oversight.

**Comment by harding** [comment]
@MarcoFalke Ah.  So far I haven't removed anything from the notes for master because of a backport.  If I do remove anything, I'll be sure that it's documented at least on the current stable branch.

Somewhat related, I was wondering if it'd be useful to tag each item in the release note with the PR or PRs that went into it so that interested readers can learn more about the context.  If I do that, it ought to at least make it easier to grep whether the release notes contain any text about a backport of PRxxxxx.

**Pull request merged**